specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/examples/PolicyStates_QueryResourceScopeExpandPolicyEvaluationDetails.json

```diff
@@ -93,12 +93,12 @@
             "policyEvaluationDetails": {
               "evaluatedExpressions": [
                 {
-                  "result": "False",
-                  "expression": "name",
-                  "path": "name",
-                  "expressionValue": "myDomainName",
-                  "targetValue": "someName",
-                  "operator": "Equals"
+                  "result": "True",
+                  "expression": "tags",
+                  "path": "tags",
+                  "expressionValue": {},
+                  "targetValue": "global-opco",
```

Comment:
This property is declared as an `object` in policyStates.json, so the example doesn't match.

Reply:
A string is also valid for the property value since it is an object.

Reply:
See https://swagger.io/docs/specification/data-models/data-types/#object - "An object is a collection of property/value pairs".

We're using Swagger 2.0, which adheres to OpenAPI 2.0 (which takes types from JSON schema draft 4). A JSON string is not valid for a "object" type field as per the JSON schema.

Reply:
I see. "any" type explained in the doc is what we need. Can I use any type?

Reply:
@praries880 can you please provide guidance on how to model a value which can be any type (string, object, array)?

Reply:
@j5lim Ill get back to you on that shortly

Reply:
@j5lim 
A good way to model this will be to break up the various properties into separate types and then define `targetValue` and `expressionValue` as composite types that encapsulate those properties...

Having a single type overloaded to support more than one type will be impossible to support in languages like `Go`

Autorest does not support the `Any` type.

Reply:
@praries880 Does it mean we need to change the RP to return `targetValue` and `expressionValue` accordingly or can it be done by swagger changes only? Do you have some example?

How about using a directive to transform to `object` type similar to what OperationalInsights spec does [here](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/operationalinsights/data-plane)? Can I do that approach?

Reply:
@praries880 Pushed a new commit where a directive is used to transform the type of the properties to `object` for the languages that support a generic `object` type. This way at least those languages can handle the response without an error. Can you please review it?